Process the mouse event if it is a click

juancnuno opened the pull request:
Hi Kirill, this is Juan with Android Studio

One of our UI tests creates an Android Instrumented Tests configuration by opening the Run/Debug Configurations dialog, clicking the +, and selecting Android Instrumented Tests in the popup. But the Swing in IDEA never gets the simulated mouse click, the instrumented test configuration is not created, and the test fails.

I traced things to 98a868fb989a64a8cdd535d1a765b7ebbd6b67a8. If I revert the commit, things work. Did you mean for that ! to be in front of isClick? The test also passes if I remove only that.

juancnuno commented:
@kkmic

kkmic commented:
Hi Juan. 
I just checked that removing isClick entirely (its calculation and from the condition) works well for the issue I was originally fixing: https://youtrack.jetbrains.com/issue/IDEA-217535
Please do so and check it your tests pass. If they do I'll remove isClick.

juancnuno commented:
Nope. Completely reverting 98a868fb989a64a8cdd535d1a765b7ebbd6b67a8 does work though.

kkmic commented:
I mean keep this code instead of the original fix:
`if (myList.locationToIndex(e.getPoint()) == myList.getSelectedIndex()) {
        super.processMouseEvent(e);
      }`

Without it one can click a disabled item such that closest enabled item executed.

juancnuno commented:
That's exactly what I went with and the test still failed. I'm not sure if you're set up to run our UI tests but it's com.android.tools.idea.tests.gui.emulator.RunInstrumentationTest if you want to experiment yourself. I also changed

```
  @NotNull
  public EditConfigurationsDialogFixture selectConfigurationType(@NotNull String confTypeName) {
    JListFixture listFixture= new JListFixture(robot(), waitForPopup(robot()));
    listFixture.replaceCellReader(CONFIGURATION_CELL_READER);
    listFixture.clickItem(confTypeName);
    return this;
  }
```

to

```
  @NotNull
  public EditConfigurationsDialogFixture selectConfigurationType(@NotNull String confTypeName) {
    JListFixture listFixture= new JListFixture(robot(), waitForPopup(robot()));
    listFixture.replaceCellReader(CONFIGURATION_CELL_READER);
    listFixture.selectItem(confTypeName);  // I changed this to selectItem
    return this;
  }
```

in com.android.tools.idea.tests.gui.framework.fixture.EditConfigurationsDialogFixture and it still failed

juancnuno commented:
Oh, let me know if you do get it running because there's another commit on our side that you'll need to get the test to pass
